### ISSUE TITLE
'--node-datadir' flag broken

### ISSUE THREAD

**Issue by ConsentDevelopment** [ethereum/mist#927]
Node type: geth
OS: windows 10 64bit
Network type: main
Mist version: 0.7.5

The blockchain data was taking a lot of room on my C: drive so I changed the location of the path C:\Users\user\AppData\Roaming (which contains the Ethereum data folder for the blockchain data) to D:\Roaming.

When I start the wallet there is a error message that it cannot find the node and so it begins to download the whole blockchain again to the path C:\Users\user\AppData\Roaming\Ethereum

As the blockchain will get bigger and bigger a lot of people will want to move the data folder to a bigger drive as the C drive is to small and has all the OS programs and data.

Solution - correctly check and use the Roaming path retrieved from the Windows OS.

* As a side note the Mist folder checks the path location correctly from the OS.

[node.log.txt](https://github.com/ethereum/mist/files/350536/node.log.txt)

**Comment by luclu** [comment]
This is possible by specifying the argument `--node-datadir` in conjunction with `--ipcpath` like this
```bash
Ethereum-Wallet.exe --node-datadir D:\Ethereum --ipcpath D:\Ethereum\geth.ipc
```

---

Unfortunately this is broken in the last versions but has been fixed on `develop/master` https://github.com/ethereum/mist/commit/83b87bf5f85fdb8f679decd4b537c7bc456cd989.

**Comment by ConsentDevelopment** [comment]
That is a solution but the wallet software should still respect the correct OS roaming path and not default to the C: drive.

**Comment by luclu** [comment]
You are right. Currently geth will query the user's home directory and expands to it: 
```javascript
home := HomeDir()
filepath.Join(home, "AppData", "Roaming", "Ethereum")
```
Could you please file an issue on the go-ethereum [repo](https://github.com/ethereum/go-ethereum/issues) for this?

**Comment by luclu** [comment]
Fixed in `0.8.1`.

**Comment by ConsentDevelopment** [comment]
Thanks. My C: SSD drive is relieved! :)
